tests/typecheck/fields/test_base.yml

```diff
@@ -150,4 +150,16 @@
                     username = models.CharField(max_length=100)
 
                 class MyModel(AuthMixin, models.Model):
-                    pass
\ No newline at end of file
+                    pass
+-   case: can_narrow_field_type
+    main: |
+        from typing import cast, NewType
+        from django.db import models
+        Year = NewType("Year", int)
+        class Book(models.Model):
+            published = cast(models.Field[Year, Year], models.IntegerField())
+        book = Book()
+        reveal_type(book.published)  # N: Revealed type is 'main.Year*'
+        book.published = 2006  # E: Incompatible types in assignment (expression has type "int", variable has type "Year")
```

Comment:
Let's also test that it can be used as an `int`, just in case.

Reply:
I mean that `book.published` can be passed to, for example, `def accepts_int(arg: int): ...`